### PR TITLE
Refactor notebook menu

### PR DESCRIPTION
**Pull request by theengineear**
Noticed an opportunity to refactor and make testing this component a little simpler and prevent bloat in what _could basically_ just be a presentation component.

### CODE REVIEW

File: packages/core/src/components/notebook-menu/extra-handlers.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  cellMap: CellMap,
+  cellIds: CellIds,
+  focusedCellId: FocusedCellId
+) => {
+  const skipToIndex = focusedCellId ? cellIds.indexOf(focusedCellId) : -1;
+  if (~skipToIndex) {
+    executeCells(executeCell, cellMap, cellIds.skip(skipToIndex));
+  } else {
+    executeCells(executeCell, cellMap, cellIds);
+  }
```
Comment (rgbkrk) [inline packages/core/src/components/notebook-menu/extra-handlers.js:45]: This all makes me wonder if we should have an action for "EXECUTE_ALL_CELLS_BELOW_FOCUSED_CELL". Same with Execute all. Force the epic to do the coordination with the state and cellMap.

**Review by rgbkrk — APPROVED** [review]
Totally cool with me, though I think we could get out of having the menu itself needing to know what the focused cell is and instead rely on an epic to handle the behavior (and know the state).

File: packages/core/src/components/notebook-menu/extra-handlers.js (hunk 1 above)
Comment (theengineear) [reply packages/core/src/components/notebook-menu/extra-handlers.js:45]: yup, totally, i just didn't want to get that deep yet

**Comment by rgbkrk** [comment]
Something I've been thinking about, regarding how we handle `execute_all` -- we could "lock the cell UI" when waiting for execute all to finish (or error), which could either be just graying it out or literally preventing edits while it's in process. Something to think about for coordinating with the epics some day in the future.

**Pull request merged**